pallets/xyk/src/lib.rs

```diff
@@ -2456,9 +2848,48 @@ impl<T: Config> XykFunctionsTrait<T::AccountId> for Pallet<T> {
 		Ok(())
 	}
 
+	fn claim_rewards_all_v2(
+		user: T::AccountId,
+		liquidity_asset_id: Self::CurrencyId,
+	) -> DispatchResult {
+		let mangata_id: TokenId = T::NativeCurrencyId::get();
+
+		let claimable_rewards =
+			Pallet::<T>::calculate_rewards_amount_v2(user.clone(), liquidity_asset_id)?;
+
+		let rewards_info: RewardInfo = Self::get_rewards_info((user.clone(), liquidity_asset_id));
+
+		let rewards_info_new: RewardInfo = RewardInfo {
+			activated_amount: rewards_info.activated_amount,
+			rewards_not_yet_claimed: 0 as u128,
+			rewards_already_claimed: rewards_info.rewards_already_claimed,
+			last_checkpoint: rewards_info.last_checkpoint,
+			pool_ratio_at_last_checkpoint: rewards_info.pool_ratio_at_last_checkpoint,
+			missing_at_last_checkpoint: rewards_info.missing_at_last_checkpoint,
+		};
+
+		RewardsInfo::<T>::insert((user.clone(), liquidity_asset_id), rewards_info_new);
```

Comment:
This should be after transfer, to prevent writing to storage before potential error. This is potentially a critical bug.
i know we are using transactional for the extrinsic that calls this, but this function can be called by thing else in the future. Perhaps we can add a require_transactionl annotation to this function.